Add crime billing online system landscape diagram

ddippolito opened the pull request:
## What does this pull request do?

**Adds "Get paid for Legal Aid - Crime" diagram**
Adds `diagrams/get-paid/get-paid-for-crime-legal-aid-system-landscape.png` diagram, titled "Crime Billing Online".

The purpose of this diagram is to show the main actors and the software components involved in the process of crime online billing, from a claim submission made by a Legal Provider to assessment and approval made by a Case Worker to the final payment issued by financial institutions. 

![crime-billing-online-system-landscape](https://media.githubusercontent.com/media/ministryofjustice/laa-architecture-documentation/get-paid-for-crime-legal-aid/diagrams/get-paid/get-paid-for-crime-legal-aid-system-landscape.png)

**Colours used on the diagram (copied from the readme)**
- Legal Aid Agency software systems, people and elements use the "Ministry of Justice" websafe colour (`#5a5c92`).
- Software systems, people and elements outside the UK Government use `#28a197`.

slorek commented:
Thanks for this. Based on my understanding of these processes, I think it might be worth adding some additional clarification:

* On the third party CMS, clarify that there are several different ones which are provided by third party software vendors
* Clarify that the third party CMS submits the claim to CCCD via its API
* Clarify that payments injected into 'CCMS Billing' is actually an invoice record, which is injected into the Open Payables interface of the Oracle Financials suite
* Clarify that the invoice is factored into the provider's weekly payment run, which includes all other payments due. It is not a distinct payment per claim.

Question: how should we explain the mechanism by which the claims are injected (i.e. CCCD > SNS > SQS > CCR/LF)? And the reciprocal feedback mechanism? It's probably too low level for this diagram, but can C4 provide a lower level view?

Question: Do caseworkers really prepare the payment in CWA themselves? I had thought this was an automatic process.

sldblog commented:
> Question: how should we explain the mechanism by which the claims are injected (i.e. CCCD > SNS > SQS > CCR/LF)? And the reciprocal feedback mechanism? It's probably too low level for this diagram, but can C4 provide a lower level view?

@slorek Yes, I think that would be a "dynamic diagram", using the https://c4model.com/ terminology. We could create one that focuses on that claim injection process.

> it would be good to say which sort of caseworker role is involved or which team they are in, as there are several different teams / roles who are referred to as caseworkers

I would find this distinction useful. We settled for "Operator" in the civil legal aid diagram and while it's kind of implicit for people _knowing_ about the context I am a bit dissatisfied with it.

sldblog commented on diagrams/get-paid/get-paid-for-crime-legal-aid-system-landscape.yaml:
```diff
@@ -0,0 +1,171 @@
+  tags: web
+  position: '2700,2200'
+- type: Software System
+  name: Integration Hub Job 1
```
*not a blocker*
Do these hub jobs have names? Would it be possible to find their location in a GitHub repo? 🤞

ddippolito replied:
I believe they do, but it'll require some investigation. As I moving towards CCR and CCLF work I can expand this

sldblog commented on diagrams/get-paid/get-paid-for-crime-legal-aid-system-landscape.yaml:
```diff
@@ -0,0 +1,171 @@
+  description: Automated Fee Calculator for AGFS fixed fee claims
+  position: '3400,900'
+- type: Software System
+  name: Means Assessment & Administration Tool - MAAT
```
👏 Love the acronym expansions! (This lazy comment applies for all of these.)

sldblog commented on diagrams/get-paid/get-paid-for-crime-legal-aid-system-landscape.yaml:
```diff
@@ -0,0 +1,171 @@
+  position: '1600,2500'
+- type: Software System
+  name: Xhibit Live
+  description: Detailed records of hearings
```
Would it be possible to link a URL for the Xhibit systems?

sldblog commented on diagrams/get-paid/get-paid-for-crime-legal-aid-system-landscape.yaml:
```diff
@@ -0,0 +1,171 @@
+  position: '4900,3000'
+- type: Software System
+  name: Laa Fee Calculator
+  description: Automated Fee Calculator for AGFS fixed fee claims
```
What is "AGFS"?

sldblog commented on diagrams/get-paid/get-paid-for-crime-legal-aid-system-landscape.yaml:
```diff
@@ -0,0 +1,171 @@
+  position: '3400,2200'
+- type: Software System
+  name: Crown Court Remuneration - CCR
+  description: LAA providers assessment tool for advocates claims
```
I don't have any context about these systems 😅
I think mentioning the final outcome for these systems would be useful.

What do you think about

> LAA providers assessment tool for advocates claims _responsible for feeding unicorns_

(but with the actual outcome :smile:)

sldblog commented on diagrams/get-paid/get-paid-for-crime-legal-aid-system-landscape.yaml:
```diff
@@ -0,0 +1,171 @@
+  position: '2725,150'
+- type: Software System
+  name: CCMS Billing
+  description: application with many responsibilities including sollicitors payments and cash collections
```
_nitpick_ _typo_ `sollicitors` &rarr; solicitors

sldblog commented on diagrams/get-paid/get-paid-for-crime-legal-aid-system-landscape.yaml:
```diff
@@ -0,0 +1,171 @@
+  tags: web,public
+  position: '2700,900'
+- type: Software System
+  name: Contract Work Administration - CWA
```
I've heard CWA mentioned as `Contracted Work and Administration` in other places. Which one would we like to use?

jsugarman commented:
@ddippolito as discussed, MAAT data is also put onto the hub nightly and CCR and CCLF copies it nightly from the hub

jsugarman commented:
And xhibit, i believe, similarly, has a job to put data in the hub which is copied to CCR/CCLF Databases nightly

ddippolito commented:
I am picking this up again, addressing some comments, adding clarification. But I'd like to get it merged and iterate on it as some of the clarification require some work. 

@slorek 

> On the third party CMS, clarify that there are several different ones which are provided by third party software vendors

Done

> Clarify that the third party CMS submits the claim to CCCD via its API
> Clarify that payments injected into 'CCMS Billing' is actually an invoice record, which is injected into the Open Payables interface of the Oracle Financials suite
> Clarify that the invoice is factored into the provider's weekly payment run, which includes all other payments due. It is not a distinct payment per claim.

Different level of abstraction. For Landscape diagrams, I'd rather not focus on how the communication happens or what technologies are involved. Different story for the cases where the Hub is involved as we're considering it a system in its own. 
@slorek @sldblog @bleach thoughts?
 
> Clarify which caseworker teams are involved and when
If that's ok, I'd rather do this as next step and merge this pull request. I need to address teams and reposibilities as a prerequesite for CCR/CCLF investigation

Also the other two questions: 

>Question: how should we explain the mechanism by which the claims are injected (i.e. CCCD > SNS > SQS > CCR/LF)? And the reciprocal feedback mechanism? It's probably too low level for this diagram, but can C4 provide a lower level view?

Yes as per @sldblog answer 🙇 

>Question: Do caseworkers really prepare the payment in CWA themselves? I had thought this was an automatic process as part of the final defence cost calculation which is triggered within CCR/LF and fed into MAAT.

Once I understand (and document) how the interaction between CCR/LF and CWA work I'll return to this. It's a whole epic imho on its own.

ddippolito commented:
@bleach 

> could you expand the CMS acronym? (I assume it's Case Management System, but not 100% sure)

Done


> it would be good to say which sort of caseworker role is involved or which team they are in, as there are several different teams / roles who are referred to as caseworkers

If that's ok, I'd rather do this as next step and merge this pull request. I need to address teams and responsibilities as a prerequisite for CCR/CCLF investigation
